R/check-cols-complete.R

```diff
@@ -23,20 +23,25 @@ check_cols_complete <- function(data, required_cols,
   if (is.null(data)) {
     return(NULL)
   }
-  ## Check if columns in `required_cols` contain missing data
+  req_cols <- required_cols[required_cols %in% names(data)]
+  missing_cols <- setdiff(required_cols, req_cols)
+
+  ## Check if columns present in data from`required_cols` contain missing data
   results <- purrr::map_lgl(
-    data[, required_cols, drop = FALSE],
+    data[, req_cols, drop = FALSE],
     function(x) any(x %in% empty_values)
   )
+
   behavior <- paste0(
     "Columns ",
     paste(required_cols, collapse = ", "),
     " should be complete."
   )
 
-  ## Return success if all required columns have complete data.
+  ## Return success if all required columns have complete data,
+  ## and there are no missing required columns.
   ## Otherwise return warn or fail depending on `strict` argument
-  if (!any(results)) {
+  if (all(!results) & length(missing_cols) == 0) {
```

Comment:
Is there a need to change `!any(results)` to `all(!results)` here?

Reply:
Nope. I changed it back to `!any(results)`, which should be a shorter runtime. Thank you for catching that!

Long, uninteresting explanation: I was getting the wrong results and was thinking it was a logic issue, not a forgetting-to-reload-package-before-testing issue. Tried out multiple different things, including changing this bit to an equivalent condition in case there was something odd with the ordering. Forgot to switch it back after I fixed the problem (i.e. reloaded the package with the newest version of the function).

Reply:
makes sense, thanks!